src/storage/store/engine.rs

```diff
@@ -826,11 +927,23 @@ impl ShardEngine {
                 .user_data_store
                 .merge(msg, txn_batch)
                 .map_err(|e| MessageValidationError::StoreError(e)),
-            MessageType::VerificationAddEthAddress | MessageType::VerificationRemove => self
+            MessageType::VerificationAddEthAddress => self
                 .stores
                 .verification_store
                 .merge(msg, txn_batch)
                 .map_err(|e| MessageValidationError::StoreError(e)),
+            MessageType::VerificationRemove => {
+                let result = self
+                    .stores
+                    .verification_store
+                    .merge(msg, txn_batch)
+                    .map_err(|e| MessageValidationError::StoreError(e));
+
+                if let Ok(hub_event) = result.as_ref() {
+                    self.handle_verification_merge_hooks(msg.fid(), hub_event, txn_batch)?;
```

Comment:
Instead otf calling `handle_verification_merge_hooks` here, I would do it one level higher. I would look at `hub_event.deleted_messages`, iterate over each of those and if any of them is a verification add then call `handle_verification_merge_hooks` on the add. This ensures that we're responding to a deleted verification add message.